project_timesheet_holidays_contract/models/hr_employee.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import datetime
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    def _get_calendar(self, date_from=None):
+        res = super()._get_calendar(date_from)
+        if not date_from:
+            return res
+
+        contracts = (
+            self.sudo()
+            .with_context(active_test=True)
+            .env["hr.contract"]
+            .search([("employee_id", "=", self.id)])
+        )
+
+        if date_from:
+            date_from = (
+                date_from if not isinstance(date_from, datetime) else date_from.date()
+            )
+            contracts = contracts.filtered(lambda c: c.date_start <= date_from)
```

Comment:
`if date_from` at line 24 will always be `True`; if not, you would've exited the function at line 15.
So you don't need this check, and you can use `date_from` directly in the `search()` domain at line 21, instead of filtering the results after.